fem/bilininteg_convection_pa.cpp

```diff
@@ -788,6 +788,15 @@ void ConvectionIntegrator::AssemblePA(const FiniteElementSpace &fes)
    {
       vel = cQ->GetVec();
    }
+   else if (VectorQuadratureFunctionCoefficient* cQ =
+               dynamic_cast<VectorQuadratureFunctionCoefficient*>(Q))
+   {
+      const QuadratureFunction &qFun = cQ->GetQuadFunction();
+      MFEM_VERIFY(qFun.Size() == dim * nq * ne,
```

Comment:
You should probably also add a check here and in the other places that makes sure the ```IntegrationRules``` are the same as well. Something along the lines of this should work:
```c++
      MFEM_VERIFY(IntRule == qFun->GetSpace()->GetElementIntRule(0),
                  "IntegrationRule used within integrator and in QuadratureFunction \
                  appear to be different");
```

Reply:
Done! I wasn't sure how to add the check, but this works nicely.